microcontroller/Core/Src/main.c

```diff
@@ -234,23 +235,25 @@ int main(void)
 		tim5.CalculationOK = 0;
 		if (!voltage_offset(pedal.first_v, pedal.second_v)
 				|| !voltage_check(pedal.user_v)) {
-			pedal.time_check += 1;
+			if (pedal.time_check < 44444) {
+				pedal.time_check += 1;
+			}
 		} else {
 			if (HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_7)) {
 				HAL_GPIO_TogglePin(GPIOC, GPIO_PIN_7); //Turn off APPS if already active
 			}
 			if (pedal.time_check > 0) {
 				pedal.time_check -= 1;
+				pedal.time_check_true = 0;
 			}
 			if (pedal.time_check == 0) {
-				pedal.time_check_true = 0;
+				pedal.time_check_true = 1;
 			}
 		}
 
 		if (time_fault_check(pedal.time_check) && pedal.time_check_true) {
-			// convert_rpm(raw value goes here)
-
-			// throttle = lookup_table(convert_throttle_input(adc_throttle_buf), steering_wheel_angle_to_steering_angle(adc_steering_buf));
+			throttle = lookup_linear(get_steering_angle_smooth(), convert_rpm(RPM));
+			throttle = throttle_percent(get_throttle_in_smooth());
```

Comment:
This doesn't seem right. Line 256 overwrites the result from line 255.

Reply:
Furthermore, `throttle_percent` seems to be converting from the raw uint32_t to a float, but `get_throttle_in_smooth()` already returns the converted float.